test/suites/scale/provisioning_test.go

```diff
@@ -0,0 +1,149 @@
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+		}})
+		provisioner = test.Provisioner(test.ProvisionerOptions{
+			ProviderRef: &v1alpha5.MachineTemplateRef{
+				Name: nodeTemplate.Name,
+			},
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1alpha5.LabelCapacityType,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{v1alpha1.CapacityTypeOnDemand},
+				},
+				{
+					Key:      v1.LabelOSStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{string(v1.Linux)},
+				},
+			},
+			// No limits!!!
+			// https://tenor.com/view/chaos-gif-22919457
+			Limits: v1.ResourceList{},
+		})
+		deployment = test.Deployment(test.DeploymentOptions{
+			PodOptions: test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("100m"),
+						v1.ResourceMemory: resource.MustParse("100Mi"),
+					},
+				},
+				TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
+			},
+		})
+		selector = labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels)
+		// Get the DS pod count and use it to calculate the DS pod overhead
+		dsCount = env.GetDaemonSetCount(provisioner)
+	})
+	It("should scale successfully on a node-dense scale-up", func(_ context.Context) {
+		replicasPerNode := 1
+		expectedNodeCount := 500
+		replicas := replicasPerNode * expectedNodeCount
+
+		deployment.Spec.Replicas = lo.ToPtr[int32](int32(replicas))
+		// Hostname anti-affinity to require one pod on each node
+		deployment.Spec.Template.Spec.Affinity = &v1.Affinity{
+			PodAntiAffinity: &v1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+					{
+						LabelSelector: deployment.Spec.Selector,
+						TopologyKey:   v1.LabelHostname,
+					},
+				},
+			},
+		}
+
+		By("waiting for the deployment to deploy all of its pods")
+		env.ExpectCreated(deployment)
+		env.EventuallyExpectPendingPodCount(selector, replicas)
+
+		env.MeasureDurationFor(func() {
+			By("kicking off provisioning by applying the provisioner and nodeTemplate")
+			env.ExpectCreated(provisioner, nodeTemplate)
+
+			env.EventuallyExpectCreatedMachineCount("==", expectedNodeCount)
+			env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectHealthyPodCount(selector, replicas)
+		}, aws.ProvisioningEventType, testGroup, "pod-dense")
+	}, SpecTimeout(10*time.Minute))
+	It("should scale successfully on a pod-dense scale-up", func(_ context.Context) {
+		replicasPerNode := 110
+		maxPodDensity := replicasPerNode + dsCount
+		expectedNodeCount := 60
+		replicas := replicasPerNode * expectedNodeCount
```

Comment:
nit: if we want to make it exactly 110 pods on the node, regardless of dsCount, we could do something like this. Don't think it matters much though.
```suggestion
                 maxPodDensity := 110
		replicasPerNode := maxPodDensity - dsCount
		expectedNodeCount := 60
		replicas := replicasPerNode * expectedNodeCount
```

Reply:
I think this makes the "math" of how many nodes that we are going to have spun up a bit more complicated. I'm fine with going slightly over 110 for this test-case. The 110 pods is more of a guideline anyways and going slightly over it doesn't seem like it would cause much of a problem.